tests/functional/s/self/self_cls_assignment.py

```diff
@@ -45,3 +50,14 @@ def _set_param(param):
 
         _set_param(param)
         return self
+
+
+class TestTypingCast:
+    """Test class for use of typing.cast"""
+
+    def function(self):
+        """This function uses typing.cast to reassign self"""
+
+        self = cast(TestTypingCast, self)
```

Comment:
Why would we cast to ``TestTypingCast`` here ? Wouldn't it be used in a more complex context ?

Reply:
I just used `self`'s class again because a similar example caused me to make this change. We can change it to something else, perhaps even a more generic `Type` if you want!